dragonchain/webserver/lib/transactions.py

```diff
@@ -108,14 +109,18 @@ def get_transaction_v1(transaction_id: str, parse: bool = True) -> Dict[str, Any
     return txn
 
 
-def submit_transaction_v1(transaction: Dict[str, Any], callback_url: Optional[str] = None) -> Dict[str, str]:
+def submit_transaction_v1(transaction: Dict[str, Any], callback_url: Optional[str], api_key: "api_key_model.APIKeyModel") -> Dict[str, str]:
     """Formats and enqueues individual transaction (from user input) to the webserver queue
     Returns:
         String of the submitted transaction id
     """
     _log.info("[TRANSACTION] Parsing and loading user input")
     txn_model = _generate_transaction_model(transaction)
 
+    # Check if allowed to create transaction of this type
+    if not api_key.is_key_allowed("transactions", "create", "create_transaction", False, extra_data={"requested_types": {txn_model.txn_type}}):
```

Comment:
should probably have a test around this.

Reply:
done